### PR TITLE
Dịch "seq2seq_vn" - Phần 4

### LINKED ISSUE #1713: Dịch "seq2seq_vn" - Phần 4
Phần cần dịch: https://github.com/aivivn/d2l-vn/blame/master/chapter_recurrent-modern/seq2seq_vn.md#L248-L329

### Hướng dẫn
- [Dịch trực tiếp trên website](https://github.com/aivivn/d2l-vn/blob/master/CONTRIBUTING.md).
- Sau khi nhấn vào phần cần dịch ở trên, hãy chọn **Normal view** và làm theo hướng dẫn ở trên để có thể bắt đầu dịch.

### PR DESCRIPTION
**Pull request by nguyenduydu**
## Pull Request Checklist

**Đánh dấu tick (sau khi "Create Pull Request") vào các mục dưới đây:**

* [x] Pull Request này tương ứng với issue nào? Trả lời số issue sau dấu `#` (không có dấu cách): Close #1713

* [x] Bản dịch này có bám sát **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)** không? Nếu một từ/cụm chưa có trong Bảng thuật ngữ, hãy tạo một Pull Request riêng đề xuất cách dịch từ/cụm đó.

* [x] Format của bản gốc có được giữ nguyên không?
    * Các phần in nghiêng, in đậm.
    * Tiêu đề (số lượng dấu `#` đầu dòng).
    * Bảng biểu, chú thích cho bảng (dòng phía trên mỗi bảng bắt đầu bằng dấu `:`).
    * Dịch các từ trong hình vẽ nếu cần. Các dòng có hình có dạng: `caption`.
    * Dịch các chú thích hình vẽ (thay các cụm `*dịch chú thích ảnh phía trên*` bằng bản dịch tiếng Việt).
    * Không dịch các phần code (nằm giữa hai dấu `````).
    * Copy các công thức toán từ bản gốc (các đoạn có `$`).
    * Giữ các dòng gán nhãn (bắt đầu với `:label:`).
    * Không tự thêm bớt các dòng trắng.
     
* [x] Trong một đoạn văn, mỗi câu văn đã được viết trong một dòng, giữa các dòng không có dòng trắng. Markdown sẽ coi những dòng liền nhau không có dòng trắng là một đoạn văn. Việc này giúp công đoạn review được thuận tiện hơn.

* [x] Tất cả chữ trong các hình vẽ đã được dịch. Nếu chưa, vui lòng xem thêm **[hướng dẫn hoặc yêu cầu dịch hình](https://github.com/aivivn/d2l-vn/blob/master/img/README.md)**.

* [x] Điền tên của bạn vào mục "Những người thực hiện" ở cuối file.

* [x] **Sau khi được approve**, điền tên những reviewers mà bạn thấy có nhiều đóng góp cho Pull Request này. Xem danh sách Contributors **[tại đây](https://github.com/aivivn/d2l-vn/blob/master/docs/contributors_info.md)**.

## Với Reviewer

1. Chỉ review nếu Pull Request này đã sẵn sàng cho việc review.

2. Đã đọc kỹ hướng dẫn đóng góp và xem qua **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)**.

3. **Bạn có thể đề xuất thay đổi cách dịch mỗi dòng trực tiếp như trong hình dưới đây:**
img

### CODE REVIEW

File: chapter_recurrent-modern/seq2seq_vn.md [hunk 1]
```diff
@@ -318,7 +327,8 @@ For a sanity check, we create identical three sequences, keep 4 elements for the
 Then the first example loss should be 2 times larger than the second one, and the last loss should be 0.
 -->
 
-*dịch đoạn phía trên*
+Để kiểm tra, ta tạo ba chuỗi giống hệt nhau, giữ 4 phần tử cho chuỗi thứ nhất, 2 phần tử cho chuỗi thứ hai và không phần tử nào cho chuỗi cuối cùng. <!-- Bạn nào review gợi ý giúp mình cụm "For a sanity check" nhé. Many thanks! -->
```
Comment (cuongvng) [inline chapter_recurrent-modern/seq2seq_vn.md:330]:
```suggestion
Để kiểm tra sơ bộ, ta tạo ba chuỗi giống hệt nhau, giữ 4 phần tử cho chuỗi thứ nhất, 2 phần tử cho chuỗi thứ hai và không phần tử nào cho chuỗi cuối cùng. <!-- Bạn nào review gợi ý giúp mình cụm "For a sanity check" nhé. Many thanks! -->
```
Trên slack có 1 thread trong channel `glossary` về từ này, em thấy mn vote là `kiểm tra sơ bộ`.

Comment (nguyenduydu) [reply chapter_recurrent-modern/seq2seq_vn.md:330]: Cám ơn Cường nhé :)

File: chapter_recurrent-modern/seq2seq_vn.md [hunk 2]
```diff
@@ -255,23 +255,27 @@ out.shape, len(state), state[0].shape, state[1].shape
...
+Lưu ý rằng ta đã đệm các câu đích để làm cho chúng có độ dài bằng nhau, nhưng ta không cần tính mất mát trên các ký tự đệm.
 
 <!--
 To implement the loss function that filters out some entries, we will use an operator called `SequenceMask`.
 It can specify to mask the first dimension (`axis=0`) or the second one (`axis=1`).
 If the second one is chosen, given a valid length vector `len` and 2-dim input `X`, this operator sets `X[i, len[i]:] = 0` for all $i$'s.
 -->
 
-*dịch đoạn phía trên*
+Để lập trình hàm mất mát để lọc ra một số phần tử, ta sẽ sử dụng một toán tử được gọi là `SequenceMask`.
```
Comment (minhduc0711) [inline chapter_recurrent-modern/seq2seq_vn.md:276]:
```suggestion
Để lập trình hàm mất mát có khả năng lọc ra một số phần tử, ta sẽ sử dụng một toán tử được gọi là `SequenceMask`.
```

File: chapter_recurrent-modern/seq2seq_vn.md [hunk 3]
```diff
@@ -255,23 +255,27 @@ out.shape, len(state), state[0].shape, state[1].shape
...
+Nó có thể chỉ định gán mặt nạ cho chiều thứ nhất (`axis=0`) hoặc thứ hai (`axis=1`).
```
Comment (minhduc0711) [inline chapter_recurrent-modern/seq2seq_vn.md:277]:
```suggestion
Nó có thể được chỉ định để gán mặt nạ cho chiều thứ nhất (`axis=0`) hoặc thứ hai (`axis=1`).
```

File: chapter_recurrent-modern/seq2seq_vn.md [hunk 4]
```diff
@@ -283,7 +287,8 @@ Apply to $n$-dim tensor $X$, it sets `X[i, len[i]:, :, ..., :] = 0`.
 In addition, we can specify the filling value such as $-1$ as shown below.
 -->
 
-*dịch đoạn phía trên*
+Áp dụng vào một tensor $n$-chiều $X$, nó sẽ đặt `X[i, len[i]:, :, ..., :] = 0`.
+Ta cũng có thể đặt một giá trị điền khác ví dụ như $-1$ như dưới đây.
```
Comment (minhduc0711) [inline chapter_recurrent-modern/seq2seq_vn.md:291]:
```suggestion
Ta cũng có thể đặt một giá trị điền khác, ví dụ $-1$, như dưới đây.
```

File: chapter_recurrent-modern/seq2seq_vn.md [hunk 5]
```diff
@@ -297,7 +302,11 @@ Then we can just use a zero weight for each example we would like to remove.
 So our customized loss function accepts an additional `valid_len` argument to ignore some failing elements in each sequence.
 -->
 
-*dịch đoạn phía trên*
+
+Bây giờ ta có thể lập trình phiên bản mặt nạ của hàm mất mát entropy chéo softmax.
```
Comment (minhduc0711) [inline chapter_recurrent-modern/seq2seq_vn.md:306]: tự nhiên giờ mình nghe từ `mặt nạ` cứ thấy kiểu gì ý :)))
```suggestion
Bây giờ ta có thể lập trình phiên bản có mặt nạ của hàm mất mát entropy chéo softmax.
```

File: chapter_recurrent-modern/seq2seq_vn.md [hunk 6]
```diff
@@ -297,7 +302,11 @@ Then we can just use a zero weight for each example we would like to remove.
...
+Lưu ý rằng hàm mất mát Gluon cho phép đặt trọng số cho mỗi mẫu, theo mặc định thì giá trị này bằng 1.
+Để loại bỏ các mẫu, ta có thể đặt trọng số cho chúng bằng 0.
```
Comment (minhduc0711) [inline chapter_recurrent-modern/seq2seq_vn.md:308]:
```suggestion
Để loại bỏ một vài mẫu nhất định, ta có thể đặt trọng số cho chúng bằng 0.
```

File: chapter_recurrent-modern/seq2seq_vn.md [hunk 7]
```diff
@@ -318,7 +327,8 @@ For a sanity check, we create identical three sequences, keep 4 elements for the
 Then the first example loss should be 2 times larger than the second one, and the last loss should be 0.
 -->
 
-*dịch đoạn phía trên*
+Để kiểm tra sơ bộ, ta tạo ba chuỗi giống hệt nhau, giữ 4 phần tử cho chuỗi thứ nhất, 2 phần tử cho chuỗi thứ hai và không phần tử nào cho chuỗi cuối cùng. <!-- Bạn nào review gợi ý giúp mình cụm "For a sanity check" nhé. Many thanks! -->
```
Comment (minhduc0711) [inline chapter_recurrent-modern/seq2seq_vn.md:330]: mình thấy `kiểm tra sợ bộ` ổn rồi

File: chapter_recurrent-modern/seq2seq_vn.md [hunk 8]
```diff
@@ -318,7 +327,8 @@ For a sanity check, we create identical three sequences, keep 4 elements for the
...
+Khi đó, mất mát của chuỗi đầu tiên phải lớn hơn 2 lần so với chuỗi thứ hai và mất mát của chuỗi cuối cùng phải bằng 0.
```
Comment (minhduc0711) [inline chapter_recurrent-modern/seq2seq_vn.md:331]:
```suggestion
Khi đó, giá trị mất mát của chuỗi đầu tiên phải lớn gấp 2 lần so với chuỗi thứ hai, còn giá trị mất mát của chuỗi cuối cùng phải bằng 0.
```

File: chapter_recurrent-modern/seq2seq_vn.md (hunk 5 above)
Comment (nguyenduydu) [reply chapter_recurrent-modern/seq2seq_vn.md:306]: Mình cũng thấy từ `mặt nạ` sao sao nhưng không biết từ nào khác :)

**Comment by nguyenduydu** [comment]
Cảm ơn @minhduc0711 đã góp ý nhé!
